process-engine/bpmn-studio
# 🏗Refactor oidc Authentication

**Changes:**

This PR Introduces big changes to the overall behavior on login with identity server.
It is the first step to refactor the authentication mechanism in BPMN-Studio and allows us to use Electron 4 from now on. 

1. Replace old login/logout flow with a new one, suitable for Electron 4.
2. Update Electron to v4.0.1.
3. Fixes necessary Parts for new Electron version.
4. Add Credentials to every remote Solution. (Subject to change)

**Issues:**

Closes #702 
Closes #705

PR: #1295

## How can others test the changes?

1. Reinstall all node modules.
3. Get the identity server: `https://github.com/process-engine/identity_server`.
3. Start the identity server with `dotnet restore` and `dotnet run`.
4. Start BPMN-Studio.
5. Click the Login Button. (A second window should open now)
6. Login
7. See all requests against remote solutions have a real access_token now.
8. Logout.
9. See that all requests use the dummy token again.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
